### ISSUE TITLE
Sync rules

### ISSUE THREAD

**Issue by rugk** [Lusito/forget-me-not#128]
AFAIK the rules are not synchronized yet. (technically said, you store sessions in the local storage API and not sync storage)

IMHO, this should change. (at least, there should be an option)

**Comment by Lusito** [comment]
Yes, this is something I've wanted to do for some time, but haven't gotten around to do yet.
Sadly, it's not as easy as just using the sync storage, as there are some things that should not be synced and some things are questionable (i.e. which state do I use when 2 devices change something at nearly the same time).

I thought about making something like a synced clipboard where you can store configurations and apply them. But currently there are more pressing issues to be completed.

**Comment by rugk** [comment]
> that should not be synced

Examples? :thinking: 

> (i.e. which state do I use when 2 devices change something at nearly the same time).

I guess that's not your issue. You use the sync storage and it returns you the latest data. I think, you do not even know (unless implemented manually), who/which device synced it or so.

> I thought about making something like a synced clipboard where you can store configurations and apply them.

So like uBlock e.g.? While totally possible, I'd be totally fine with just using sync storage by default and sync it all, if needed. I mean, I see no use-case why a thing should not be synced…

**Comment by Lusito** [comment]
>Examples?

- The list of domains to clean on startup.
- Settings that might want to be set differently depending on device type. Like "show notification", since notifications on android are quite annoying.
- The last tab name (if you configured FMN to remember the last tab you had open).

**Comment by virhonestum** [comment]
I would also abosultely love this feature!
Here is my suggestion: Add a toggle to disable browser sync. When sync is disabled, copy the settings to the local storage and use local storage. When sync is re-enabled copy it to sync storage and use that.
All rules should be synced, including the ones at startup. The two more device specific settings like last tab name and show notifications should not be synced. This can be easily disabledm but it's much more work to re-add all your rules than to just re-edit a few small checkboxes.

**Comment by Lusito** [comment]
>All rules should be synced, including the ones at startup.

I guess this is in response to my earlier "The list of domains to clean on startup."?
In that case: I was not talking about rules. Since cleanup happens on startup rather than on shutdown, I need to remember domains that had been open before closing the browser. That's the list I was talking about.

Aside from that: It seems like storage.sync is not implemented on android yet, so it would only get synced between desktop browsers: https://bugzilla.mozilla.org/show_bug.cgi?id=1316442

Not sure how much use this feature is without android support. Are there many people using multiple firefox desktop browsers? I see the main use-case in desktop <-> smartphone syncing.

**Comment by virhonestum** [comment]
Oh, okay, that's what you mean, well in my opinion this also seems very device specific, so no sync for that. 
I am using Firefox on my desktop computer as well as on my laptop, additionally I totally wipe and re-install my laptop every few months, so for both usecases a sync would greatly reduce the workload. It's just much simpler to automatically download the data isntead of having to copy the profile folders over every time something changed.

**Comment by syrkuit** [comment]
I expect desktop<->desktop to be plenty useful, anyone with a home & work desktop is likely to want it for example.

**Comment by Nugman** [comment]
I would love to see this feature. I just discovered Forget Me Not and use it to manage my privacy on the internet. Of course it's pretty tiring to add rules for whitelists on 3 computers (work, private workstation and notebook).

**Comment by cordafab** [comment]
> I thought about making something like a synced clipboard where you can store configurations and apply them.

This would be awesome!

**Comment by Rik44444** [comment]
yes please build in sync for rules!! I use firefox on 4 different devices and keep adding rules:p

**Comment by yfede** [comment]
Perhaps, for a start, you could add two buttons, one to "save current rules to firefox account" and one to "load rules from firefox account"

The latter, could even be avoided if the extension can determine that there is a new "saved state" and updates automatically from that.
 
My point is, that this way it's the user that decides when the rules are ready to be "committed" to all its other devices (desktops for now), partly solving his potential confusion if the sync happens always automatically, sometimes with unintended settings.

And also, this would give the chance to prompt the user for what settings he wants to save in the cloud, giving him control for not overriding some (excuse me if this sounds vague, I've just started using this extension)
